src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,868 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+import "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+import "@openzeppelin-upgrades/contracts/security/ReentrancyGuardUpgradeable.sol";
+
+import "../permissions/Pausable.sol";
+import "../libraries/EIP1271SignatureUtils.sol";
+import "../libraries/SlashingConstants.sol";
+import "./AllocationManagerStorage.sol";
+
+contract AllocationManager is
+    Initializable,
+    OwnableUpgradeable,
+    Pausable,
+    AllocationManagerStorage,
+    ReentrancyGuardUpgradeable
+{
+    using Snapshots for Snapshots.History;
+
+    /// @dev BIPS factor for slashable bips
+    uint256 internal constant BIPS_FACTOR = 10_000;
```

Comment:
can we move this definition to something like libraries/SlashingConstants.sol ?

Reply:
Using WAD in SlashingLib now